classes/Device.class.php

```diff
@@ -1526,6 +1526,24 @@ function SearchDevicebySerialNo(){
 		return $deviceList;
 	}
 
+	function SearchDevicebySerialNoExact(){
```

Comment:
Instead of adding a new function here, we should instead reuse the function above this but add an exact option.

https://github.com/opendcim/openDCIM/blob/0ea5e3c0c172945d34675bd21de99ceb1dddc3bb/classes/Device.class.php#L1513

```
	function SearchDevicebySerialNo($exact=false){
		global $dbh;

		$this->MakeSafe();

		if($exact){
			$extend="SerialNo = :serial LIMIT 1;"
		}else{
			$extend="SerialNo LIKE \"%$this->SerialNo%\" ORDER BY Label;"
		}

		$sql="SELECT * FROM fac_Device WHERE $extend";

		$deviceList=array();

		foreach($dbh->query($sql) as $deviceRow){
			$deviceList[$deviceRow["DeviceID"]]=Device::RowToObject($deviceRow);
		}

		return $deviceList;
	}
```

then over there in devices it would become $devList = $dev->SearchDevicebySerialNo(true);